pkgs/tools/networking/curl/default.nix

```diff
@@ -133,7 +134,7 @@ stdenv.mkDerivation (finalAttrs: {
       "--without-ca-path"
     ] ++ lib.optionals (!gnutlsSupport && !opensslSupport && !wolfsslSupport && !rustlsSupport) [
       "--without-ssl"
-    ];
+    ] ++ lib.optionals ipv6Support [ "--enable-ipv6" ];
```

Comment:
use
```nix
(lib.enableFeature ipv6Support "ipv6")
```
instead so when `ipv6Support` is false `--disable-ipv6` will be passed to configure. seems like the different platforms have different ipv6 defaults.